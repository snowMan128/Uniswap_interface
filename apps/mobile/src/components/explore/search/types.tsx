import { SearchResult } from 'wallet/src/features/search/SearchResult'
import { SEARCH_RESULT_HEADER_KEY } from './constants'

// Header type used to render header text instead of SearchResult item

export type SearchResultOrHeader =
  | SearchResult
  | { type: typeof SEARCH_RESULT_HEADER_KEY; title: string }
