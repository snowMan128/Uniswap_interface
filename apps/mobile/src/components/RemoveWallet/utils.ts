import { CommonActions } from '@react-navigation/core'
import { dispatchNavigationAction } from 'src/app/navigation/rootNavigation'
import { OnboardingScreens, Screens } from 'src/screens/Screens'
import { ImportType, OnboardingEntryPoint } from 'wallet/src/features/onboarding/types'

// This fast-forwards user to the same app state as if
// they have pressed "Get Started" on Landing and should now see import method view
export function navigateToOnboardingImportMethod(): void {
  dispatchNavigationAction(
    CommonActions.reset({
      index: 0,
      routes: [
        {
          name: Screens.OnboardingStack,
          state: {
            index: 1,
            routes: [
              {
                name: OnboardingScreens.Landing,
                params: {
                  entryPoint: OnboardingEntryPoint.FreshInstallOrReplace,
                  importType: ImportType.NotYetSelected,
                },
              },
              {
                name: OnboardingScreens.ImportMethod,
                params: {
                  entryPoint: OnboardingEntryPoint.FreshInstallOrReplace,
                  importType: ImportType.NotYetSelected,
                },
              },
            ],
          },
        },
      ],
    })
  )
}
