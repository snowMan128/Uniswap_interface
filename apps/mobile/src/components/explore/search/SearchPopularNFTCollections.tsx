import React, { useMemo } from 'react'
import { FlatList, ListRenderItemInfo } from 'react-native'
import { SearchNFTCollectionItem } from 'src/components/explore/search/items/SearchNFTCollectionItem'
import {
  getSearchResultId,
  gqlNFTToNFTCollectionSearchResult,
} from 'src/components/explore/search/utils'
import { Inset } from 'ui/src'
import { TokenLoader } from 'wallet/src/components/loading/TokenLoader'
import { useSearchPopularNftCollectionsQuery } from 'wallet/src/data/__generated__/types-and-hooks'
import {
  NFTCollectionSearchResult,
  SearchResultType,
} from 'wallet/src/features/search/SearchResult'

function isNFTCollectionSearchResult(
  result: NFTCollectionSearchResult | null
): result is NFTCollectionSearchResult {
  return (result as NFTCollectionSearchResult).type === SearchResultType.NFTCollection
}

export function SearchPopularNFTCollections(): JSX.Element {
  // Load popular NFTs by top trading volume
  const { data, loading } = useSearchPopularNftCollectionsQuery()

  const formattedItems = useMemo(() => {
    if (!data?.topCollections?.edges) {
      return
    }

    const searchResults = data.topCollections.edges.map(({ node }) =>
      gqlNFTToNFTCollectionSearchResult(node)
    )
    return searchResults.filter(isNFTCollectionSearchResult)
  }, [data])

  if (loading) {
    return (
      <Inset all="$spacing8">
        <TokenLoader repeat={2} />
      </Inset>
    )
  }

  return (
    <FlatList
      data={formattedItems}
      keyExtractor={getSearchResultId}
      renderItem={renderNFTCollectionItem}
    />
  )
}

const renderNFTCollectionItem = ({
  item,
}: ListRenderItemInfo<NFTCollectionSearchResult>): JSX.Element => (
  <SearchNFTCollectionItem collection={item} />
)
