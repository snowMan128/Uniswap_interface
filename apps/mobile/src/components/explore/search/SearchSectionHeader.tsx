import React from 'react'
import { Flex, Text, TextProps } from 'ui/src'

interface SectionHeaderTextProps {
  title: string
  icon?: JSX.Element
}

export const SectionHeaderText = ({
  title,
  icon,
  ...rest
}: SectionHeaderTextProps & TextProps): JSX.Element => {
  return (
    <Flex row alignItems="center" gap="$spacing8" mb="$spacing4" mx="$spacing4" {...rest}>
      {icon && icon}
      <Text color="$neutral2" variant="subheading2">
        {title}
      </Text>
    </Flex>
  )
}
