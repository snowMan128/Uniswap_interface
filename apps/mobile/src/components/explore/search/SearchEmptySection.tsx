import React, { useMemo } from 'react'
import { useTranslation } from 'react-i18next'
import { FlatList } from 'react-native'
import { FadeIn, FadeOut } from 'react-native-reanimated'
import { useAppDispatch, useAppSelector } from 'src/app/hooks'
import { SearchPopularNFTCollections } from 'src/components/explore/search/SearchPopularNFTCollections'
import { SearchPopularTokens } from 'src/components/explore/search/SearchPopularTokens'
import { renderSearchItem } from 'src/components/explore/search/SearchResultsSection'
import { SectionHeaderText } from 'src/components/explore/search/SearchSectionHeader'
import { AnimatedFlex, Flex, Text, TouchableArea, useSporeColors } from 'ui/src'
import ClockIcon from 'ui/src/assets/icons/clock.svg'
import TrendArrowIcon from 'ui/src/assets/icons/trend-up.svg'
import { iconSizes } from 'ui/src/theme'
import { FEATURE_FLAGS } from 'wallet/src/features/experiments/constants'
import { useFeatureFlag } from 'wallet/src/features/experiments/hooks'
import { clearSearchHistory } from 'wallet/src/features/search/searchHistorySlice'
import {
  SearchResult,
  SearchResultType,
  WalletSearchResult,
} from 'wallet/src/features/search/SearchResult'
import { selectSearchHistory } from 'wallet/src/features/search/selectSearchHistory'

export const SUGGESTED_WALLETS: WalletSearchResult[] = [
  {
    type: SearchResultType.ENSAddress,
    address: '0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045',
    ensName: 'vitalik.eth',
  },
  {
    type: SearchResultType.ENSAddress,
    address: '0x50EC05ADe8280758E2077fcBC08D878D4aef79C3',
    ensName: 'hayden.eth',
  },
]

export function SearchEmptySection(): JSX.Element {
  const { t } = useTranslation()
  const dispatch = useAppDispatch()
  const searchHistory = useAppSelector(selectSearchHistory)
  const unitagFeatureFlagEnabled = useFeatureFlag(FEATURE_FLAGS.Unitags)

  const onPressClearSearchHistory = (): void => {
    dispatch(clearSearchHistory())
  }

  const modifiedHistory: SearchResult[] = useMemo(
    () =>
      searchHistory.map((historyItem: SearchResult) => {
        if (!unitagFeatureFlagEnabled && historyItem.type === SearchResultType.Unitag) {
          return {
            type: SearchResultType.WalletByAddress,
            address: historyItem.address,
            searchId: historyItem.searchId,
          }
        } else {
          return historyItem
        }
      }),
    [searchHistory, unitagFeatureFlagEnabled]
  )

  // Show search history (if applicable), trending tokens, and wallets
  return (
    <AnimatedFlex entering={FadeIn} exiting={FadeOut} gap="$spacing12">
      {searchHistory.length > 0 && (
        <AnimatedFlex entering={FadeIn} exiting={FadeOut}>
          <FlatList
            ListHeaderComponent={
              <Flex
                row
                alignItems="center"
                gap="$spacing16"
                justifyContent="space-between"
                mb="$spacing4">
                <SectionHeaderText
                  icon={<RecentIcon />}
                  title={t('explore.search.section.recent')}
                />
                <TouchableArea onPress={onPressClearSearchHistory}>
                  <Text color="$accent1" variant="buttonLabel3">
                    {t('explore.search.action.clear')}
                  </Text>
                </TouchableArea>
              </Flex>
            }
            data={modifiedHistory}
            renderItem={(props): JSX.Element | null =>
              renderSearchItem({ ...props, searchContext: { isHistory: true } })
            }
          />
        </AnimatedFlex>
      )}
      <Flex gap="$spacing4">
        <SectionHeaderText icon={<TrendIcon />} title={t('explore.search.section.popularTokens')} />
        <SearchPopularTokens />
      </Flex>
      <Flex gap="$spacing4">
        <SectionHeaderText icon={<TrendIcon />} title={t('explore.search.section.popularNFT')} />
        <SearchPopularNFTCollections />
      </Flex>
      <FlatList
        ListHeaderComponent={
          <SectionHeaderText
            icon={<TrendIcon />}
            title={t('explore.search.section.suggestedWallets')}
          />
        }
        data={SUGGESTED_WALLETS}
        keyExtractor={walletKey}
        renderItem={renderSearchItem}
      />
    </AnimatedFlex>
  )
}

const walletKey = (wallet: WalletSearchResult): string => {
  return wallet.address
}

export const TrendIcon = (): JSX.Element => {
  const colors = useSporeColors()
  return <TrendArrowIcon color={colors.neutral2.get()} width={iconSizes.icon20} />
}

export const RecentIcon = (): JSX.Element => {
  const colors = useSporeColors()
  return (
    <ClockIcon color={colors.neutral2.get()} height={iconSizes.icon20} width={iconSizes.icon20} />
  )
}
