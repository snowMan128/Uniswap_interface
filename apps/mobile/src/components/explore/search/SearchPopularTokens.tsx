import React, { useMemo } from 'react'
import { FlatList, ListRenderItemInfo } from 'react-native'
import { SearchTokenItem } from 'src/components/explore/search/items/SearchTokenItem'
import { getSearchResultId } from 'src/components/explore/search/utils'
import { Inset } from 'ui/src'
import { TokenLoader } from 'wallet/src/components/loading/TokenLoader'
import { fromGraphQLChain } from 'wallet/src/features/chains/utils'
import { SearchResultType, TokenSearchResult } from 'wallet/src/features/search/SearchResult'
import { TopToken, usePopularTokens } from 'wallet/src/features/tokens/hooks'

function gqlTokenToTokenSearchResult(token: Maybe<TopToken>): TokenSearchResult | null {
  if (!token || !token.project) {
    return null
  }

  const { chain, address, symbol, project } = token
  const { name } = project
  const chainId = fromGraphQLChain(chain)
  if (!chainId || !symbol || !name) {
    return null
  }

  return {
    type: SearchResultType.Token,
    chainId,
    address,
    name,
    symbol,
    logoUrl: project?.logoUrl,
  } as TokenSearchResult
}

export function SearchPopularTokens(): JSX.Element {
  const { popularTokens, loading } = usePopularTokens()
  const tokens = useMemo(
    () =>
      popularTokens
        ?.map(gqlTokenToTokenSearchResult)
        .filter((t): t is TokenSearchResult => Boolean(t)),
    [popularTokens]
  )

  if (loading) {
    return (
      <Inset all="$spacing8">
        <TokenLoader repeat={2} />
      </Inset>
    )
  }

  return <FlatList data={tokens} keyExtractor={getSearchResultId} renderItem={renderTokenItem} />
}

const renderTokenItem = ({ item }: ListRenderItemInfo<TokenSearchResult>): JSX.Element => (
  <SearchTokenItem token={item} />
)
