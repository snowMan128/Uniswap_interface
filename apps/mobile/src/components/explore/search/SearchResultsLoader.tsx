import React from 'react'
import { useTranslation } from 'react-i18next'
import { FadeIn, FadeOut } from 'react-native-reanimated'
import { SectionHeaderText } from 'src/components/explore/search/SearchSectionHeader'
import { AnimatedFlex, Flex } from 'ui/src'
import { TokenLoader } from 'wallet/src/components/loading/TokenLoader'

export const SearchResultsLoader = (): JSX.Element => {
  const { t } = useTranslation()
  return (
    <Flex gap="$spacing16">
      <Flex gap="$spacing12">
        <SectionHeaderText title={t('Tokens')} />
        <AnimatedFlex entering={FadeIn} exiting={FadeOut} mx="$spacing8">
          <TokenLoader repeat={2} />
        </AnimatedFlex>
      </Flex>
      <Flex gap="$spacing12">
        <SectionHeaderText title={t('NFT Collections')} />
        <AnimatedFlex entering={FadeIn} exiting={FadeOut} mx="$spacing8">
          <TokenLoader repeat={2} />
        </AnimatedFlex>
      </Flex>
      <Flex gap="$spacing12">
        <SectionHeaderText title={t('Wallets')} />
        <AnimatedFlex entering={FadeIn} exiting={FadeOut} mx="$spacing8">
          <TokenLoader />
        </AnimatedFlex>
      </Flex>
    </Flex>
  )
}
