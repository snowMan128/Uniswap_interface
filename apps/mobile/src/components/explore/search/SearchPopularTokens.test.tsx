import React from 'react'
import { SearchPopularTokens } from 'src/components/explore/search/SearchPopularTokens'
import { render, screen } from 'src/test/test-utils'
import { ethToken, usdcToken, wethToken } from 'wallet/src/test/fixtures'
import { queryResolvers } from 'wallet/src/test/utils'

const { resolvers } = queryResolvers({
  topTokens: () => [wethToken(), usdcToken()],
  tokens: () => [ethToken({ address: null })],
})

describe(SearchPopularTokens, () => {
  it('renders without error', async () => {
    const tree = render(<SearchPopularTokens />, { resolvers })

    // Loading should show Token loader
    expect(screen.getAllByText('Token Full Name')).toBeDefined()
    expect(tree.toJSON()).toMatchSnapshot()

    // Success where WETH result in topTokens is replaced by ETH
    expect(await screen.findByText('ETH')).toBeDefined()
    expect(screen.getByText('USDC')).toBeDefined()
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
