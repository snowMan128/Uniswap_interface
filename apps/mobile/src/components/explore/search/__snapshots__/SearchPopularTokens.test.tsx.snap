// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPopularTokens renders without error 1`] = `
<View
  style={
    {
      "flexDirection": "column",
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      {
        "flexDirection": "column",
        "opacity": 0,
      }
    }
    testID="shimmer-placeholder"
  >
    <View
      style={
        {
          "flexDirection": "column",
          "flexGrow": 1,
          "gap": 4,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 56,
            "opacity": 1,
            "paddingBottom": 8,
            "paddingTop": 8,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "flexGrow": 1,
              "gap": 12,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#CECECE",
                "borderBottomLeftRadius": 999999,
                "borderBottomRightRadius": 999999,
                "borderTopLeftRadius": 999999,
                "borderTopRightRadius": 999999,
                "flexDirection": "column",
                "minHeight": 40,
                "minWidth": 40,
              }
            }
          />
          <View
            style={
              {
                "alignItems": "flex-start",
                "flexDirection": "column",
                "flexGrow": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
              testID="text-placeholder"
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "position": "relative",
                  }
                }
              >
                <View
                  accessibilityElementsHidden={true}
                  importantForAccessibility="no-hide-descendants"
                >
                  <Text
                    allowFontScaling={true}
                    maxFontSizeMultiplier={1.4}
                    numberOfLines={1}
                    style={
                      {
                        "color": "transparent",
                        "fontFamily": "Basel-Book",
                        "fontSize": 19,
                        "lineHeight": 24,
                        "opacity": 0,
                      }
                    }
                    suppressHighlighting={true}
                  >
                    Token Full Name
                  </Text>
                </View>
                <View
                  style={
                    {
                      "backgroundColor": "#F9F9F9",
                      "borderBottomLeftRadius": 4,
                      "borderBottomRightRadius": 4,
                      "borderTopLeftRadius": 4,
                      "borderTopRightRadius": 4,
                      "bottom": "5%",
                      "flexDirection": "column",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": "5%",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "gap": 8,
                  "minHeight": 20,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
                testID="text-placeholder"
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "position": "relative",
                    }
                  }
                >
                  <View
                    accessibilityElementsHidden={true}
                    importantForAccessibility="no-hide-descendants"
                  >
                    <Text
                      allowFontScaling={true}
                      maxFontSizeMultiplier={1.4}
                      numberOfLines={1}
                      style={
                        {
                          "color": "transparent",
                          "fontFamily": "Basel-Book",
                          "fontSize": 17,
                          "lineHeight": 24,
                          "opacity": 0,
                        }
                      }
                      suppressHighlighting={true}
                    >
                      1,000 TFN
                    </Text>
                  </View>
                  <View
                    style={
                      {
                        "backgroundColor": "#F9F9F9",
                        "borderBottomLeftRadius": 4,
                        "borderBottomRightRadius": 4,
                        "borderTopLeftRadius": 4,
                        "borderTopRightRadius": 4,
                        "bottom": "5%",
                        "flexDirection": "column",
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": "5%",
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 56,
            "opacity": 0.5,
            "paddingBottom": 8,
            "paddingTop": 8,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "flexGrow": 1,
              "gap": 12,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#CECECE",
                "borderBottomLeftRadius": 999999,
                "borderBottomRightRadius": 999999,
                "borderTopLeftRadius": 999999,
                "borderTopRightRadius": 999999,
                "flexDirection": "column",
                "minHeight": 40,
                "minWidth": 40,
              }
            }
          />
          <View
            style={
              {
                "alignItems": "flex-start",
                "flexDirection": "column",
                "flexGrow": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
              testID="text-placeholder"
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "position": "relative",
                  }
                }
              >
                <View
                  accessibilityElementsHidden={true}
                  importantForAccessibility="no-hide-descendants"
                >
                  <Text
                    allowFontScaling={true}
                    maxFontSizeMultiplier={1.4}
                    numberOfLines={1}
                    style={
                      {
                        "color": "transparent",
                        "fontFamily": "Basel-Book",
                        "fontSize": 19,
                        "lineHeight": 24,
                        "opacity": 0,
                      }
                    }
                    suppressHighlighting={true}
                  >
                    Token Full Name
                  </Text>
                </View>
                <View
                  style={
                    {
                      "backgroundColor": "#F9F9F9",
                      "borderBottomLeftRadius": 4,
                      "borderBottomRightRadius": 4,
                      "borderTopLeftRadius": 4,
                      "borderTopRightRadius": 4,
                      "bottom": "5%",
                      "flexDirection": "column",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": "5%",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "gap": 8,
                  "minHeight": 20,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
                testID="text-placeholder"
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "position": "relative",
                    }
                  }
                >
                  <View
                    accessibilityElementsHidden={true}
                    importantForAccessibility="no-hide-descendants"
                  >
                    <Text
                      allowFontScaling={true}
                      maxFontSizeMultiplier={1.4}
                      numberOfLines={1}
                      style={
                        {
                          "color": "transparent",
                          "fontFamily": "Basel-Book",
                          "fontSize": 17,
                          "lineHeight": 24,
                          "opacity": 0,
                        }
                      }
                      suppressHighlighting={true}
                    >
                      1,000 TFN
                    </Text>
                  </View>
                  <View
                    style={
                      {
                        "backgroundColor": "#F9F9F9",
                        "borderBottomLeftRadius": 4,
                        "borderBottomRightRadius": 4,
                        "borderTopLeftRadius": 4,
                        "borderTopRightRadius": 4,
                        "bottom": "5%",
                        "flexDirection": "column",
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": "5%",
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`SearchPopularTokens renders without error 2`] = `
<RCTScrollView
  data={
    [
      {
        "address": null,
        "chainId": 1,
        "logoUrl": "https://loremflickr.com/640/480",
        "name": "Ethereum",
        "safetyLevel": "VERIFIED",
        "symbol": "ETH",
        "type": 1,
      },
      {
        "address": "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48",
        "chainId": 1,
        "logoUrl": "https://loremflickr.com/640/480",
        "name": "USD Coin",
        "safetyLevel": "VERIFIED",
        "symbol": "USDC",
        "type": 1,
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        actions={
          [
            {
              "onPress": [Function],
              "systemIcon": "heart.fill",
              "title": "Remove favorite",
            },
            {
              "onPress": [Function],
              "systemIcon": "arrow.2.squarepath",
              "title": "Swap",
            },
            {
              "onPress": [Function],
              "systemIcon": "qrcode",
              "title": "Receive",
            },
            {
              "onPress": [Function],
              "systemIcon": "square.and.arrow.up",
              "title": "Share",
            },
          ]
        }
        onPress={[Function]}
      >
        <View
          cancelable={true}
          disabled={false}
          focusable={true}
          hitSlop={[Function]}
          minPressDuration={0}
          onBlur={[Function]}
          onFocus={[Function]}
          onLongPress={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onPress={[Function]}
          onPressIn={[Function]}
          onPressOut={[Function]}
          style={
            {
              "flexDirection": "column",
              "opacity": 1,
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
          testID="search-token-item"
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "gap": 12,
                "paddingBottom": 12,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 12,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "column",
                  "height": 40,
                  "justifyContent": "center",
                  "width": 40,
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "https://loremflickr.com/640/480",
                  }
                }
                style={
                  [
                    {
                      "resizeMode": "contain",
                    },
                    {
                      "backgroundColor": "#2222220D",
                      "borderColor": "#2222220D",
                      "borderRadius": 20,
                      "borderWidth": 0.5,
                      "height": 40,
                      "width": 40,
                    },
                  ]
                }
              />
            </View>
            <View
              style={
                {
                  "alignItems": "flex-start",
                  "flexDirection": "column",
                  "flexShrink": 1,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 8,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    {
                      "flexDirection": "column",
                      "flexShrink": 1,
                    }
                  }
                >
                  <Text
                    allowFontScaling={true}
                    maxFontSizeMultiplier={1.4}
                    numberOfLines={1}
                    style={
                      {
                        "color": "#222222",
                        "fontFamily": "Basel-Book",
                        "fontSize": 19,
                        "lineHeight": 24,
                      }
                    }
                    suppressHighlighting={true}
                  >
                    Ethereum
                  </Text>
                </View>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 8,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  allowFontScaling={true}
                  maxFontSizeMultiplier={1.4}
                  numberOfLines={1}
                  style={
                    {
                      "color": "#7D7D7D",
                      "fontFamily": "Basel-Book",
                      "fontSize": 17,
                      "lineHeight": 24,
                    }
                  }
                  suppressHighlighting={true}
                >
                  ETH
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        actions={
          [
            {
              "onPress": [Function],
              "systemIcon": "heart",
              "title": "Favorite token",
            },
            {
              "onPress": [Function],
              "systemIcon": "arrow.2.squarepath",
              "title": "Swap",
            },
            {
              "onPress": [Function],
              "systemIcon": "qrcode",
              "title": "Receive",
            },
            {
              "onPress": [Function],
              "systemIcon": "square.and.arrow.up",
              "title": "Share",
            },
          ]
        }
        onPress={[Function]}
      >
        <View
          cancelable={true}
          disabled={false}
          focusable={true}
          hitSlop={[Function]}
          minPressDuration={0}
          onBlur={[Function]}
          onFocus={[Function]}
          onLongPress={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onPress={[Function]}
          onPressIn={[Function]}
          onPressOut={[Function]}
          style={
            {
              "flexDirection": "column",
              "opacity": 1,
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
          testID="search-token-item"
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "gap": 12,
                "paddingBottom": 12,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 12,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "column",
                  "height": 40,
                  "justifyContent": "center",
                  "width": 40,
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "https://loremflickr.com/640/480",
                  }
                }
                style={
                  [
                    {
                      "resizeMode": "contain",
                    },
                    {
                      "backgroundColor": "#2222220D",
                      "borderColor": "#2222220D",
                      "borderRadius": 20,
                      "borderWidth": 0.5,
                      "height": 40,
                      "width": 40,
                    },
                  ]
                }
              />
            </View>
            <View
              style={
                {
                  "alignItems": "flex-start",
                  "flexDirection": "column",
                  "flexShrink": 1,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 8,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    {
                      "flexDirection": "column",
                      "flexShrink": 1,
                    }
                  }
                >
                  <Text
                    allowFontScaling={true}
                    maxFontSizeMultiplier={1.4}
                    numberOfLines={1}
                    style={
                      {
                        "color": "#222222",
                        "fontFamily": "Basel-Book",
                        "fontSize": 19,
                        "lineHeight": 24,
                      }
                    }
                    suppressHighlighting={true}
                  >
                    USD Coin
                  </Text>
                </View>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 8,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  allowFontScaling={true}
                  maxFontSizeMultiplier={1.4}
                  numberOfLines={1}
                  style={
                    {
                      "color": "#7D7D7D",
                      "fontFamily": "Basel-Book",
                      "fontSize": 17,
                      "lineHeight": 24,
                    }
                  }
                  suppressHighlighting={true}
                >
                  USDC
                </Text>
                <View
                  style={
                    {
                      "flexDirection": "column",
                      "flexShrink": 1,
                    }
                  }
                >
                  <Text
                    allowFontScaling={true}
                    maxFontSizeMultiplier={1.4}
                    numberOfLines={1}
                    style={
                      {
                        "color": "#7D7D7D",
                        "fontFamily": "Basel-Book",
                        "fontSize": 17,
                        "lineHeight": 24,
                      }
                    }
                    suppressHighlighting={true}
                  >
                    0xa0b8...eb48
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
